src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -1,128 +1,139 @@
-#[derive(Debug, Clone, Default)]
-pub struct {{ target_kind_name }} {
-{% for field in fields -%}
-         {{ field.name | snake | indent(first = true) }}: {{ field.value_type }},
-{% endfor %}
-}
+{# Includes the macros for anonymous types that will be added as child classes #}
+{% import "anonymous_type.templ"  as macros %}
+use {{ target_kind_name | snake }}::{{ target_kind_name }};
+pub mod  {{ target_kind_name | snake }} {
+    use super::*;
 
-impl {{ target_kind_name }} {
-    pub fn new({% for field in fields | sort(attribute="name") -%}{{ field.name | snake }}: {{ field.value_type }},{% endfor %}) -> Self {
-        Self {
-            {% for field in fields -%}
-            {{ field.name | snake }},
-            {% endfor %}
-        }
+    #[derive(Debug, Clone, Default)]
```

Comment:
(🗺️ PR tour) most of the following changes are due to indentation of having structs inside a module. The only change to consider is [here](https://github.com/amazon-ion/ion-cli/pull/109/files#diff-41ca7e87bb08901ab9968e0b36925f52aac1789f104fdec039438782c2f87246R136-R138).